cmd/kubeadm/app/preflight/checks.go

```diff
@@ -605,3 +608,55 @@ func TryStartKubelet() {
 		}
 	}
 }
+
+// HostResolvesLocalCheck checks that a given hostname resovles to a local IP address.
+type HostResolvesLocalCheck struct {
+	hostname string
+}
+
+func (lc HostResolvesLocalCheck) Check() (warnings, errors []error) {
+
+	resolvedIPs, err := net.LookupIP(lc.hostname)
+	if err != nil {
+		errors = append(errors, fmt.Errorf("Could not resolve %s: %s", lc.hostname, err))
+		return nil, errors
+	}
+
+	warnings = []error{}
+
+	// Early exit if the address is a loopback address.
+	// note: this does not address the extreme corner case where lo interface is down.
+	for _, ri := range resolvedIPs {
+		if ri.IsLoopback() {
+			return warnings, errors
+		}
+	}
+
+	localIPAddrs, err := net.InterfaceAddrs()
+	if err != nil {
+		errors = append(errors, fmt.Errorf("Could not retrieve interface addresses: %s", err))
+		return nil, errors
+	}
+
+	for _, li := range localIPAddrs {
+		ip, _, err := net.ParseCIDR(li.String())
```

Comment:
`net.InterfaceAddrs()` returns addresses in CIDR notation (ie. 172.18.0.1/16), therefore we need to use `net.ParseCIDR()` here.